### PR TITLE
3.7 version checking

### PR DESCRIPTION
**Pull request by Eugene-hu**
Blacklist missed previous version checking
- Adds back version checking for older signatures
- Anything older than 3.7.0 will not be able to query the servers

### CODE REVIEW

**Review by isabella618033 — APPROVED** [review]

**Review by camfairchild — APPROVED** [review]
🔥 🚒

File: bittensor/_axon/__init__.py [hunk 1]
```diff
@@ -412,8 +412,12 @@ def parse_signature(
     ) -> Tuple[int, str, str, str, int]:
         r"""Attempts to parse a signature from the metadata"""
         signature = metadata.get("bittensor-signature")
+        version = metadata.get('bittensor-version')
         if signature is None:
             raise Exception("Request signature missing")
+        if version < 370:
+            raise Exception("Incorrect Version")
+        
         for parser in [self.parse_signature_v2, self.parse_legacy_signature]:
```
Comment (adriansmares) [inline bittensor/_axon/__init__.py:421]: You can just drop the `self.parse_legacy_signature` from below. Clients older than 3.6.0 will send such signatures and will be rejected.

File: bittensor/_axon/__init__.py [hunk 2]
```diff
@@ -412,8 +412,12 @@ def parse_signature(
     ) -> Tuple[int, str, str, str, int]:
         r"""Attempts to parse a signature from the metadata"""
         signature = metadata.get("bittensor-signature")
+        version = metadata.get('bittensor-version')
```
Comment (adriansmares) [inline bittensor/_axon/__init__.py:415]: Metadata is a `Dict[str, str]`. You need to cast to integer if you want to compare it with an integer.

Comment (Eugene-hu) [reply bittensor/_axon/__init__.py:415]: Thanks Adrian!

**Pull request merged**